ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/displayedgraph/ProcessStatus.scala

```diff
@@ -54,12 +49,17 @@ object ProcessStatus {
       errorMessage
     )
 
-  def create(processState: ProcessState, expectedDeploymentVersion: Option[Long]): ProcessStatus = {
-    val versionMatchMessage = (processState.version, expectedDeploymentVersion) match {
-      case (Some(stateVersion), Some(expectedVersion)) if stateVersion.versionId == expectedVersion => None
-      case (Some(stateVersion), Some(expectedVersion)) if processState.isDeployed => Some(s"Process deployed in version ${stateVersion.versionId} (by ${stateVersion.user}), expected version $expectedVersion")
+  def isStateAfterDeploy(processState: ProcessState): Boolean =
+    processState.status.isDuringDeploy || processState.status.isRunning || processState.status.isFinished
+
+  def create(processState: ProcessState, lastAction: Option[ProcessDeploymentAction]): ProcessStatus = {
+    //TODO: Move this logic to another place.. This should be moved together with ManagementActor.handleObsoleteStatus
+    val versionMatchMessage = (processState.processVersionId, lastAction) match {
+      case (Some(stateVersion), Some(action)) if stateVersion.versionId == action.processVersionId => None
+      case (Some(stateVersion), Some(action)) if action.isDeployed && !isStateAfterDeploy(processState) => Some(s"Process deployed in version ${stateVersion.versionId} (by ${stateVersion.user}), but currently not working")
+      case (Some(stateVersion), Some(action)) if action.isDeployed && stateVersion.versionId != action.processVersionId => Some(s"Process deployed in version ${stateVersion.versionId} (by ${stateVersion.user}), expected version ${action.processVersionId}")
       case (Some(stateVersion), None) if processState.isDeployed => Some(s"Process deployed in version ${stateVersion.versionId} (by ${stateVersion.user}), should not be deployed")
-      case _  => None
+      case _ => None
```

Comment:
What about the `case (None, Some(action))`? Why we not need to handle cases like `case (None, Some(action)) if !action.isDeployed`?

If this message is only for deployed case, WDYT about extracting it to method like `deployedVersionMismatchMessage`? and change local variable name in the same manner.

Reply:
Changed.